From 871ee2f08715f8fae441786db04ba8497822f79a Mon Sep 17 00:00:00 2001
From: Daan De Meyer <daan.j.demeyer@gmail.com>
Date: Tue, 8 Oct 2024 16:22:58 +0200
Subject: [PATCH 1/3] bus-util: Log more information when connecting to a bus
 socket fails

Let's log about which bus we're trying to connect to and what transport
we're using to do it.

(cherry picked from commit d8a77d55e6ad7f251ae0eb6758af6bba111095df)
---
 src/analyze/analyze-blame.c             |  2 +-
 src/analyze/analyze-critical-chain.c    |  2 +-
 src/analyze/analyze-dot.c               |  2 +-
 src/analyze/analyze-dump.c              |  2 +-
 src/analyze/analyze-fdstore.c           |  2 +-
 src/analyze/analyze-log-control.c       |  2 +-
 src/analyze/analyze-malloc.c            |  2 +-
 src/analyze/analyze-plot.c              |  2 +-
 src/analyze/analyze-security.c          |  2 +-
 src/analyze/analyze-service-watchdogs.c |  2 +-
 src/analyze/analyze-time.c              |  2 +-
 src/busctl/busctl.c                     |  2 +-
 src/cgls/cgls.c                         |  9 ++++-----
 src/firstboot/firstboot.c               |  4 ++--
 src/home/homectl.c                      |  2 +-
 src/hostname/hostnamectl.c              |  2 +-
 src/import/importctl.c                  |  2 +-
 src/locale/localectl.c                  |  2 +-
 src/login/inhibit.c                     |  2 +-
 src/login/loginctl.c                    |  2 +-
 src/machine/machinectl.c                |  2 +-
 src/mount/mount-tool.c                  |  2 +-
 src/portable/portablectl.c              |  2 +-
 src/run/run.c                           |  2 +-
 src/shared/bus-util.c                   | 18 ++++++++++++++----
 src/shared/bus-util.h                   |  4 +++-
 src/shared/cgroup-show.c                |  2 +-
 src/systemctl/systemctl-util.c          |  2 +-
 src/timedate/timedatectl.c              |  2 +-
 29 files changed, 48 insertions(+), 37 deletions(-)

diff --git a/src/analyze/analyze-blame.c b/src/analyze/analyze-blame.c
index 81e5c590b9cc2..4d78ed405d7d4 100644
--- a/src/analyze/analyze-blame.c
+++ b/src/analyze/analyze-blame.c
@@ -14,7 +14,7 @@ int verb_blame(int argc, char *argv[], void *userdata) {
 
         r = acquire_bus(&bus, NULL);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, arg_runtime_scope);
 
         n = acquire_time_data(bus, /* require_finished = */ false, &times);
         if (n <= 0)
diff --git a/src/analyze/analyze-critical-chain.c b/src/analyze/analyze-critical-chain.c
index 7d78de3d388d1..03c014d4023e6 100644
--- a/src/analyze/analyze-critical-chain.c
+++ b/src/analyze/analyze-critical-chain.c
@@ -200,7 +200,7 @@ int verb_critical_chain(int argc, char *argv[], void *userdata) {
 
         r = acquire_bus(&bus, NULL);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, arg_runtime_scope);
 
         n = acquire_time_data(bus, /* require_finished = */ true, &times);
         if (n <= 0)
diff --git a/src/analyze/analyze-dot.c b/src/analyze/analyze-dot.c
index 9e92d59bceeee..b93bf4a9a5bb0 100644
--- a/src/analyze/analyze-dot.c
+++ b/src/analyze/analyze-dot.c
@@ -149,7 +149,7 @@ int verb_dot(int argc, char *argv[], void *userdata) {
 
         r = acquire_bus(&bus, NULL);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, arg_runtime_scope);
 
         r = expand_patterns(bus, strv_skip(argv, 1), &expanded_patterns);
         if (r < 0)
diff --git a/src/analyze/analyze-dump.c b/src/analyze/analyze-dump.c
index 2642582903b09..4ee547ed4c969 100644
--- a/src/analyze/analyze-dump.c
+++ b/src/analyze/analyze-dump.c
@@ -123,7 +123,7 @@ int verb_dump(int argc, char *argv[], void *userdata) {
 
         r = acquire_bus(&bus, NULL);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, arg_runtime_scope);
 
         pager_open(arg_pager_flags);
 
diff --git a/src/analyze/analyze-fdstore.c b/src/analyze/analyze-fdstore.c
index 8ada6d4e73e1f..1f4adb7089ab0 100644
--- a/src/analyze/analyze-fdstore.c
+++ b/src/analyze/analyze-fdstore.c
@@ -98,7 +98,7 @@ int verb_fdstore(int argc, char *argv[], void *userdata) {
 
         r = acquire_bus(&bus, NULL);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, arg_runtime_scope);
 
         STRV_FOREACH(arg, strv_skip(argv, 1)) {
                 r = dump_fdstore(bus, *arg);
diff --git a/src/analyze/analyze-log-control.c b/src/analyze/analyze-log-control.c
index cead0e833f5da..854cf6cb806c2 100644
--- a/src/analyze/analyze-log-control.c
+++ b/src/analyze/analyze-log-control.c
@@ -12,7 +12,7 @@ int verb_log_control(int argc, char *argv[], void *userdata) {
 
         r = acquire_bus(&bus, NULL);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, arg_runtime_scope);
 
         r = verb_log_control_common(bus, "org.freedesktop.systemd1", argv[0], argc == 2 ? argv[1] : NULL);
         if (r < 0)
diff --git a/src/analyze/analyze-malloc.c b/src/analyze/analyze-malloc.c
index 5e6ff5bb9627c..514526d14256a 100644
--- a/src/analyze/analyze-malloc.c
+++ b/src/analyze/analyze-malloc.c
@@ -43,7 +43,7 @@ int verb_malloc(int argc, char *argv[], void *userdata) {
 
         r = acquire_bus(&bus, NULL);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, arg_runtime_scope);
 
         r = sd_bus_can_send(bus, SD_BUS_TYPE_UNIX_FD);
         if (r < 0)
diff --git a/src/analyze/analyze-plot.c b/src/analyze/analyze-plot.c
index e271296b6875b..ce67c092c37cd 100644
--- a/src/analyze/analyze-plot.c
+++ b/src/analyze/analyze-plot.c
@@ -468,7 +468,7 @@ int verb_plot(int argc, char *argv[], void *userdata) {
 
         r = acquire_bus(&bus, &use_full_bus);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, arg_runtime_scope);
 
         n = acquire_boot_times(bus, /* require_finished = */ true, &boot);
         if (n < 0)
diff --git a/src/analyze/analyze-security.c b/src/analyze/analyze-security.c
index 3d7b647a6f361..d3ba975107297 100644
--- a/src/analyze/analyze-security.c
+++ b/src/analyze/analyze-security.c
@@ -2904,7 +2904,7 @@ int verb_security(int argc, char *argv[], void *userdata) {
         if (!arg_offline) {
                 r = acquire_bus(&bus, NULL);
                 if (r < 0)
-                        return bus_log_connect_error(r, arg_transport);
+                        return bus_log_connect_error(r, arg_transport, arg_runtime_scope);
         }
 
         pager_open(arg_pager_flags);
diff --git a/src/analyze/analyze-service-watchdogs.c b/src/analyze/analyze-service-watchdogs.c
index 6535eb1a894f7..b1fad4328caa5 100644
--- a/src/analyze/analyze-service-watchdogs.c
+++ b/src/analyze/analyze-service-watchdogs.c
@@ -16,7 +16,7 @@ int verb_service_watchdogs(int argc, char *argv[], void *userdata) {
 
         r = acquire_bus(&bus, NULL);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, arg_runtime_scope);
 
         if (argc == 1) {
                 /* get ServiceWatchdogs */
diff --git a/src/analyze/analyze-time.c b/src/analyze/analyze-time.c
index c233b1f08558e..8135112dec42b 100644
--- a/src/analyze/analyze-time.c
+++ b/src/analyze/analyze-time.c
@@ -11,7 +11,7 @@ int verb_time(int argc, char *argv[], void *userdata) {
 
         r = acquire_bus(&bus, NULL);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, arg_runtime_scope);
 
         r = pretty_boot_time(bus, &buf);
         if (r < 0)
diff --git a/src/busctl/busctl.c b/src/busctl/busctl.c
index 8db9076997232..bbd36c1dc42bc 100644
--- a/src/busctl/busctl.c
+++ b/src/busctl/busctl.c
@@ -152,7 +152,7 @@ static int acquire_bus(bool set_monitor, sd_bus **ret) {
 
         r = sd_bus_start(bus);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, arg_runtime_scope);
 
         *ret = TAKE_PTR(bus);
 
diff --git a/src/cgls/cgls.c b/src/cgls/cgls.c
index 70fa260246929..1dcd316cdb379 100644
--- a/src/cgls/cgls.c
+++ b/src/cgls/cgls.c
@@ -221,13 +221,12 @@ static int run(int argc, char *argv[]) {
                                         return log_error_errno(r, "Failed to mangle unit name: %m");
 
                                 if (!bus) {
+                                        RuntimeScope scope = arg_show_unit == SHOW_UNIT_USER ? RUNTIME_SCOPE_USER : RUNTIME_SCOPE_SYSTEM;
+
                                         /* Connect to the bus only if necessary */
-                                        r = bus_connect_transport_systemd(
-                                                        BUS_TRANSPORT_LOCAL, NULL,
-                                                        arg_show_unit == SHOW_UNIT_USER ? RUNTIME_SCOPE_USER : RUNTIME_SCOPE_SYSTEM,
-                                                        &bus);
+                                        r = bus_connect_transport_systemd(BUS_TRANSPORT_LOCAL, NULL, scope, &bus);
                                         if (r < 0)
-                                                return bus_log_connect_error(r, BUS_TRANSPORT_LOCAL);
+                                                return bus_log_connect_error(r, BUS_TRANSPORT_LOCAL, scope);
                                 }
 
                                 q = show_cgroup_get_unit_path_and_warn(bus, unit_name, &cgroup);
diff --git a/src/firstboot/firstboot.c b/src/firstboot/firstboot.c
index 0dbdfc663871f..3b732b38609ab 100644
--- a/src/firstboot/firstboot.c
+++ b/src/firstboot/firstboot.c
@@ -1616,7 +1616,7 @@ static int reload_system_manager(sd_bus **bus) {
         if (!*bus) {
                 r = bus_connect_transport_systemd(BUS_TRANSPORT_LOCAL, NULL, RUNTIME_SCOPE_SYSTEM, bus);
                 if (r < 0)
-                        return bus_log_connect_error(r, BUS_TRANSPORT_LOCAL);
+                        return bus_log_connect_error(r, BUS_TRANSPORT_LOCAL, RUNTIME_SCOPE_SYSTEM);
         }
 
         r = bus_service_manager_reload(*bus);
@@ -1639,7 +1639,7 @@ static int reload_vconsole(sd_bus **bus) {
         if (!*bus) {
                 r = bus_connect_transport_systemd(BUS_TRANSPORT_LOCAL, NULL, RUNTIME_SCOPE_SYSTEM, bus);
                 if (r < 0)
-                        return bus_log_connect_error(r, BUS_TRANSPORT_LOCAL);
+                        return bus_log_connect_error(r, BUS_TRANSPORT_LOCAL, RUNTIME_SCOPE_SYSTEM);
         }
 
         r = bus_wait_for_jobs_new(*bus, &w);
diff --git a/src/home/homectl.c b/src/home/homectl.c
index 346daa1195ccd..36e3bf729948f 100644
--- a/src/home/homectl.c
+++ b/src/home/homectl.c
@@ -134,7 +134,7 @@ static int acquire_bus(sd_bus **bus) {
 
         r = bus_connect_transport(arg_transport, arg_host, RUNTIME_SCOPE_SYSTEM, bus);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, RUNTIME_SCOPE_SYSTEM);
 
         (void) sd_bus_set_allow_interactive_authorization(*bus, arg_ask_password);
 
diff --git a/src/hostname/hostnamectl.c b/src/hostname/hostnamectl.c
index d1c4d476f66b9..ead0cc4da6e82 100644
--- a/src/hostname/hostnamectl.c
+++ b/src/hostname/hostnamectl.c
@@ -813,7 +813,7 @@ static int run(int argc, char *argv[]) {
 
         r = bus_connect_transport(arg_transport, arg_host, RUNTIME_SCOPE_SYSTEM, &bus);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, RUNTIME_SCOPE_SYSTEM);
 
         return hostnamectl_main(bus, argc, argv);
 }
diff --git a/src/import/importctl.c b/src/import/importctl.c
index f939d80815dca..da9e5b62d23d6 100644
--- a/src/import/importctl.c
+++ b/src/import/importctl.c
@@ -1235,7 +1235,7 @@ static int run(int argc, char *argv[]) {
 
         r = bus_connect_transport(arg_transport, arg_host, RUNTIME_SCOPE_SYSTEM, &bus);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, RUNTIME_SCOPE_SYSTEM);
 
         (void) sd_bus_set_allow_interactive_authorization(bus, arg_ask_password);
 
diff --git a/src/locale/localectl.c b/src/locale/localectl.c
index 32354027f1755..2fb4657fe24fb 100644
--- a/src/locale/localectl.c
+++ b/src/locale/localectl.c
@@ -527,7 +527,7 @@ static int run(int argc, char *argv[]) {
 
         r = bus_connect_transport(arg_transport, arg_host, RUNTIME_SCOPE_SYSTEM, &bus);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, RUNTIME_SCOPE_SYSTEM);
 
         return localectl_main(bus, argc, argv);
 }
diff --git a/src/login/inhibit.c b/src/login/inhibit.c
index 4682830d198a8..e060c614004b1 100644
--- a/src/login/inhibit.c
+++ b/src/login/inhibit.c
@@ -265,7 +265,7 @@ static int run(int argc, char *argv[]) {
 
         r = sd_bus_default_system(&bus);
         if (r < 0)
-                return bus_log_connect_error(r, BUS_TRANSPORT_LOCAL);
+                return bus_log_connect_error(r, BUS_TRANSPORT_LOCAL, RUNTIME_SCOPE_SYSTEM);
 
         if (arg_action == ACTION_LIST)
                 return print_inhibitors(bus);
diff --git a/src/login/loginctl.c b/src/login/loginctl.c
index cf3bff437a0bc..e950bfc985ff5 100644
--- a/src/login/loginctl.c
+++ b/src/login/loginctl.c
@@ -1727,7 +1727,7 @@ static int run(int argc, char *argv[]) {
 
         r = bus_connect_transport(arg_transport, arg_host, RUNTIME_SCOPE_SYSTEM, &bus);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, RUNTIME_SCOPE_SYSTEM);
 
         (void) sd_bus_set_allow_interactive_authorization(bus, arg_ask_password);
 
diff --git a/src/machine/machinectl.c b/src/machine/machinectl.c
index 1b63e6d203786..b9105f298b6b1 100644
--- a/src/machine/machinectl.c
+++ b/src/machine/machinectl.c
@@ -2446,7 +2446,7 @@ static int run(int argc, char *argv[]) {
 
         r = bus_connect_transport(arg_transport, arg_host, RUNTIME_SCOPE_SYSTEM, &bus);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, RUNTIME_SCOPE_SYSTEM);
 
         (void) sd_bus_set_allow_interactive_authorization(bus, arg_ask_password);
 
diff --git a/src/mount/mount-tool.c b/src/mount/mount-tool.c
index fcebdcaf18c9b..b991bed714896 100644
--- a/src/mount/mount-tool.c
+++ b/src/mount/mount-tool.c
@@ -1503,7 +1503,7 @@ static int run(int argc, char* argv[]) {
 
         r = bus_connect_transport_systemd(arg_transport, arg_host, arg_runtime_scope, &bus);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, arg_runtime_scope);
 
         if (arg_action == ACTION_UMOUNT)
                 return action_umount(bus, argc, argv);
diff --git a/src/portable/portablectl.c b/src/portable/portablectl.c
index 57b930d6cba41..2681cf4dd2070 100644
--- a/src/portable/portablectl.c
+++ b/src/portable/portablectl.c
@@ -229,7 +229,7 @@ static int acquire_bus(sd_bus **bus) {
 
         r = bus_connect_transport(arg_transport, arg_host, RUNTIME_SCOPE_SYSTEM, bus);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, RUNTIME_SCOPE_SYSTEM);
 
         (void) sd_bus_set_allow_interactive_authorization(*bus, arg_ask_password);
 
diff --git a/src/run/run.c b/src/run/run.c
index ba7bb2148a15f..3a7102e32923b 100644
--- a/src/run/run.c
+++ b/src/run/run.c
@@ -2405,7 +2405,7 @@ static int run(int argc, char* argv[]) {
         else
                 r = bus_connect_transport_systemd(arg_transport, arg_host, arg_runtime_scope, &bus);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, arg_runtime_scope);
 
         if (arg_scope)
                 return start_transient_scope(bus);
diff --git a/src/shared/bus-util.c b/src/shared/bus-util.c
index 30f9602b1edbc..f347ea604ec25 100644
--- a/src/shared/bus-util.c
+++ b/src/shared/bus-util.c
@@ -28,6 +28,7 @@
 #include "path-util.h"
 #include "socket-util.h"
 #include "stdio-util.h"
+#include "string-table.h"
 #include "uid-classification.h"
 
 static int name_owner_change_callback(sd_bus_message *m, void *userdata, sd_bus_error *ret_error) {
@@ -49,14 +50,14 @@ int bus_log_address_error(int r, BusTransport transport) {
                                       "Failed to set bus address: %m");
 }
 
-int bus_log_connect_error(int r, BusTransport transport) {
+int bus_log_connect_error(int r, BusTransport transport, RuntimeScope scope) {
         bool hint_vars = transport == BUS_TRANSPORT_LOCAL && r == -ENOMEDIUM,
              hint_addr = transport == BUS_TRANSPORT_LOCAL && ERRNO_IS_PRIVILEGE(r);
 
         return log_error_errno(r,
-                               r == hint_vars ? "Failed to connect to bus: $DBUS_SESSION_BUS_ADDRESS and $XDG_RUNTIME_DIR not defined (consider using --machine=<user>@.host --user to connect to bus of other user)" :
-                               r == hint_addr ? "Failed to connect to bus: Operation not permitted (consider using --machine=<user>@.host --user to connect to bus of other user)" :
-                                                "Failed to connect to bus: %m");
+                               r == hint_vars ? "Failed to connect to %s scope bus via %s transport: $DBUS_SESSION_BUS_ADDRESS and $XDG_RUNTIME_DIR not defined (consider using --machine=<user>@.host --user to connect to bus of other user)" :
+                               r == hint_addr ? "Failed to connect to %s scope bus via %s transport: Operation not permitted (consider using --machine=<user>@.host --user to connect to bus of other user)" :
+                                                "Failed to connect to %s scope bus via %s transport: %m", runtime_scope_to_string(scope), bus_transport_to_string(transport));
 }
 
 int bus_async_unregister_and_exit(sd_event *e, sd_bus *bus, const char *name) {
@@ -926,3 +927,12 @@ int bus_message_read_id128(sd_bus_message *m, sd_id128_t *ret) {
                 return -EINVAL;
         }
 }
+
+static const char* const bus_transport_table[] = {
+        [BUS_TRANSPORT_LOCAL]   = "local",
+        [BUS_TRANSPORT_REMOTE]  = "remote",
+        [BUS_TRANSPORT_MACHINE] = "machine",
+        [BUS_TRANSPORT_CAPSULE] = "capsule",
+};
+
+DEFINE_STRING_TABLE_LOOKUP_TO_STRING(bus_transport, BusTransport);
diff --git a/src/shared/bus-util.h b/src/shared/bus-util.h
index 9c6f01d8eb846..fbccb24314fe4 100644
--- a/src/shared/bus-util.h
+++ b/src/shared/bus-util.h
@@ -48,7 +48,7 @@ int bus_connect_transport(BusTransport transport, const char *host, RuntimeScope
 int bus_connect_transport_systemd(BusTransport transport, const char *host, RuntimeScope runtime_scope, sd_bus **bus);
 
 int bus_log_address_error(int r, BusTransport transport);
-int bus_log_connect_error(int r, BusTransport transport);
+int bus_log_connect_error(int r, BusTransport transport, RuntimeScope scope);
 
 #define bus_log_parse_error(r)                                  \
         log_error_errno(r, "Failed to parse bus message: %m")
@@ -84,3 +84,5 @@ int bus_creds_get_pidref(sd_bus_creds *c, PidRef *ret);
 int bus_query_sender_pidref(sd_bus_message *m, PidRef *ret);
 
 int bus_message_read_id128(sd_bus_message *m, sd_id128_t *ret);
+
+const char* bus_transport_to_string(BusTransport transport) _const_;
diff --git a/src/shared/cgroup-show.c b/src/shared/cgroup-show.c
index 87177316da8f9..96826ac7d8e87 100644
--- a/src/shared/cgroup-show.c
+++ b/src/shared/cgroup-show.c
@@ -433,7 +433,7 @@ int show_cgroup_get_path_and_warn(
 
                 r = bus_connect_transport_systemd(BUS_TRANSPORT_LOCAL, NULL, RUNTIME_SCOPE_SYSTEM, &bus);
                 if (r < 0)
-                        return bus_log_connect_error(r, BUS_TRANSPORT_LOCAL);
+                        return bus_log_connect_error(r, BUS_TRANSPORT_LOCAL, RUNTIME_SCOPE_SYSTEM);
 
                 r = show_cgroup_get_unit_path_and_warn(bus, unit, &root);
                 if (r < 0)
diff --git a/src/systemctl/systemctl-util.c b/src/systemctl/systemctl-util.c
index 38e1f23740ed6..18c8823e3fe70 100644
--- a/src/systemctl/systemctl-util.c
+++ b/src/systemctl/systemctl-util.c
@@ -54,7 +54,7 @@ int acquire_bus(BusFocus focus, sd_bus **ret) {
                 else
                         r = bus_connect_transport(arg_transport, arg_host, arg_runtime_scope, &buses[focus]);
                 if (r < 0)
-                        return bus_log_connect_error(r, arg_transport);
+                        return bus_log_connect_error(r, arg_transport, arg_runtime_scope);
 
                 (void) sd_bus_set_allow_interactive_authorization(buses[focus], arg_ask_password);
         }
diff --git a/src/timedate/timedatectl.c b/src/timedate/timedatectl.c
index 46ec6b31bc46a..db9f3b22b9b5e 100644
--- a/src/timedate/timedatectl.c
+++ b/src/timedate/timedatectl.c
@@ -1031,7 +1031,7 @@ static int run(int argc, char *argv[]) {
 
         r = bus_connect_transport(arg_transport, arg_host, RUNTIME_SCOPE_SYSTEM, &bus);
         if (r < 0)
-                return bus_log_connect_error(r, arg_transport);
+                return bus_log_connect_error(r, arg_transport, RUNTIME_SCOPE_SYSTEM);
 
         return timedatectl_main(bus, argc, argv);
 }

From 31e38b55b2e4bb1aa42fe106ea14df8e82758303 Mon Sep 17 00:00:00 2001
From: Daan De Meyer <daan.j.demeyer@gmail.com>
Date: Tue, 8 Oct 2024 16:25:52 +0200
Subject: [PATCH 2/3] core: Bump log level of reexecute request to notice

A daemon-reload is important enough to deserve logging at notice
level.

(cherry picked from commit 4ee41be82507348fbbc9d3ab28aae6330eb51663)
---
 src/core/dbus-manager.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/core/dbus-manager.c b/src/core/dbus-manager.c
index 7da35a803c62c..2412bd1aab742 100644
--- a/src/core/dbus-manager.c
+++ b/src/core/dbus-manager.c
@@ -1571,10 +1571,10 @@ static void log_caller(sd_bus_message *message, Manager *manager, const char *me
         (void) sd_bus_creds_get_comm(creds, &comm);
         caller = manager_get_unit_by_pid(manager, pid);
 
-        log_info("%s requested from client PID " PID_FMT "%s%s%s%s%s%s...",
-                 method, pid,
-                 comm ? " ('" : "", strempty(comm), comm ? "')" : "",
-                 caller ? " (unit " : "", caller ? caller->id : "", caller ? ")" : "");
+        log_notice("%s requested from client PID " PID_FMT "%s%s%s%s%s%s...",
+                   method, pid,
+                   comm ? " ('" : "", strempty(comm), comm ? "')" : "",
+                   caller ? " (unit " : "", caller ? caller->id : "", caller ? ")" : "");
 }
 
 static int method_reload(sd_bus_message *message, void *userdata, sd_bus_error *error) {

From 814be7116dda14074749253d94b83387ceff0ff1 Mon Sep 17 00:00:00 2001
From: Daan De Meyer <daan.j.demeyer@gmail.com>
Date: Tue, 8 Oct 2024 16:28:25 +0200
Subject: [PATCH 3/3] core: Log in more scenarios about which process initiated
 an operation

Exit/Reboot/Poweroff and similar operations are invasive enough that
logging about who initiated them is very useful to debug issues.

(cherry picked from commit acb0f501f4291efce82bcf89d4ad92b6a895f4fa)
---
 src/core/dbus-manager.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/core/dbus-manager.c b/src/core/dbus-manager.c
index 2412bd1aab742..0c1260f2d9cbe 100644
--- a/src/core/dbus-manager.c
+++ b/src/core/dbus-manager.c
@@ -1671,6 +1671,8 @@ static int method_exit(sd_bus_message *message, void *userdata, sd_bus_error *er
         if (r < 0)
                 return r;
 
+        log_caller(message, m, "Exit");
+
         /* Exit() (in contrast to SetExitCode()) is actually allowed even if
          * we are running on the host. It will fall back on reboot() in
          * systemd-shutdown if it cannot do the exit() because it isn't a
@@ -1695,6 +1697,8 @@ static int method_reboot(sd_bus_message *message, void *userdata, sd_bus_error *
                 return sd_bus_error_setf(error, SD_BUS_ERROR_NOT_SUPPORTED,
                                          "Reboot is only supported for system managers.");
 
+        log_caller(message, m, "Reboot");
+
         m->objective = MANAGER_REBOOT;
 
         return sd_bus_reply_method_return(message, NULL);
@@ -1737,6 +1741,8 @@ static int method_soft_reboot(sd_bus_message *message, void *userdata, sd_bus_er
                         return -ENOMEM;
         }
 
+        log_caller(message, m, "Soft reboot");
+
         free_and_replace(m->switch_root, rt);
         m->objective = MANAGER_SOFT_REBOOT;
 
@@ -1757,6 +1763,8 @@ static int method_poweroff(sd_bus_message *message, void *userdata, sd_bus_error
                 return sd_bus_error_setf(error, SD_BUS_ERROR_NOT_SUPPORTED,
                                          "Powering off is only supported for system managers.");
 
+        log_caller(message, m, "Poweroff");
+
         m->objective = MANAGER_POWEROFF;
 
         return sd_bus_reply_method_return(message, NULL);
@@ -1776,6 +1784,8 @@ static int method_halt(sd_bus_message *message, void *userdata, sd_bus_error *er
                 return sd_bus_error_setf(error, SD_BUS_ERROR_NOT_SUPPORTED,
                                          "Halt is only supported for system managers.");
 
+        log_caller(message, m, "Halt");
+
         m->objective = MANAGER_HALT;
 
         return sd_bus_reply_method_return(message, NULL);
@@ -1795,6 +1805,8 @@ static int method_kexec(sd_bus_message *message, void *userdata, sd_bus_error *e
                 return sd_bus_error_setf(error, SD_BUS_ERROR_NOT_SUPPORTED,
                                          "KExec is only supported for system managers.");
 
+        log_caller(message, m, "Kexec");
+
         m->objective = MANAGER_KEXEC;
 
         return sd_bus_reply_method_return(message, NULL);
